### PR TITLE
Add VIS.X adapter

### PR DESCRIPTION
**Pull request by mk0x9**

### CODE REVIEW

**Comment by mansinahar** [comment]
@mk0x9 Looks like there are some conflicts. Can you please resolve them and update the PR when you have a chance? Thanks!

**Comment by mk0x9** [comment]
@mansinahar I have rebased branch on current master HEAD.

**Comment by mansinahar** [comment]
Thanks @mk0x9. Appreciate it :)

File: adapters/visx/visxtest/params/race/banner.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mansinahar) [inline adapters/visx/visxtest/params/race/banner.json]: I believe you need to specify sample test params here rather than keeping it empty since you do expect Visx specific params to be sent in the request

Comment (mk0x9) [reply adapters/visx/visxtest/params/race/banner.json]: Added them

Comment (mansinahar) [reply adapters/visx/visxtest/params/race/banner.json]: Thanks! Also, looks like there's some new conflicts unfortunately. Can you please resolve them?

**Comment by josephveach** [comment]
You should rebase off current master branch.  This will eliminate the conflicts that are currently occurring due to another adapter that was submitted since you first pulled source.  Once you rebase, you will need to add a new argument to _MakeRequests()_ in  adapters/visx/visx.go:

MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo)

**Comment by mansinahar** [comment]
@mk0x9 Pinging to see if you have a chance to resolve the merge conflicts

File: exchange/adapter_map.go [hunk 2]
```diff
@@ -69,6 +70,7 @@ func newAdapterMap(client *http.Client, cfg *config.Configuration, infos adapter
 		openrtb_ext.BidderGrid:           grid.NewGridBidder(cfg.Adapters[string(openrtb_ext.BidderGrid)].Endpoint),
 		openrtb_ext.BidderSonobi:         sonobi.NewSonobiBidder(client, cfg.Adapters[string(openrtb_ext.BidderSonobi)].Endpoint),
```
Comment (guscarreon) [inline exchange/adapter_map.go]: Add this line here to resolve merging conflicts:
```
              openrtb_ext.BidderUnruly:         unruly.NewUnrulyBidder(client, cfg.Adapters[string(openrtb_ext.BidderUnruly)].Endpoint),
```

File: config/config.go [hunk 3]
```diff
@@ -446,6 +446,7 @@ func (cfg *Configuration) setDerivedDefaults() {
 	setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderYieldmo, "https://ads.yieldmo.com/pbsync?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&redirectUri="+url.QueryEscape(externalURL)+"%2Fsetuid%3Fbidder%3Dyieldmo%26gdpr%3D{{.GDPR}}%26gdpr_consent%3D{{.GDPRConsent}}%26uid%3D%24UID")
 	setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderGamoshi, "https://rtb.gamoshi.io/pix/0000/scm?gdpr={{.GDPR}}&consent={{.GDPRConsent}}&rurl="+url.QueryEscape(externalURL)+"%2Fsetuid%3Fbidder%3Dgamoshi%26gdpr%3D{{.GDPR}}%26gdpr_consent%3D{{.GDPRConsent}}%26uid%3D%5Bgusr%5D")
 	setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderMgid, "https://cm.mgid.com/m?cdsp=363893&adu="+url.QueryEscape(externalURL)+"%2Fsetuid%3Fbidder%3Dmgid%26gdpr%3D{{.GDPR}}%26gdpr_consent%3D{{.GDPRConsent}}%26uid%3D%7Bmuidn%7D")
```
Comment (guscarreon) [inline config/config.go]: Add this line here to resolve merge conflicts:
```
      setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderUnruly, "https://usermatch.targeting.unrulymedia.com/pbsync?gdpr={{.GDPR}}&consent={{.GDPRConsent}}&rurl="+url.QueryEscape(externalURL)+"%2Fsetuid%3Fbidder%3Dunruly%26gdpr%3D{{.GDPR}}%26gdpr_consent%3D{{.GDPRConsent}}%26uid%3D%24UID")
```

File: config/config.go [hunk 4]
```diff
@@ -604,6 +605,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.yieldmo.endpoint", "http://ads.yieldmo.com/exchange/prebid-server")
```
Comment (guscarreon) [inline config/config.go]: Add this line here to resolve merge conflicts:
```
	v.SetDefault("adapters.unruly.endpoint", "http://targeting.unrulymedia.com/openrtb/2.2")
```

File: usersync/usersyncers/syncer.go [hunk 5]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/prebid/prebid-server/adapters/somoaudience"
 	"github.com/prebid/prebid-server/adapters/sonobi"
 	"github.com/prebid/prebid-server/adapters/sovrn"
```
Comment (guscarreon) [inline usersync/usersyncers/syncer.go]: Add this line to resolve merge conflicts:
```
      "github.com/prebid/prebid-server/adapters/unruly"
```

File: usersync/usersyncers/syncer.go [hunk 6]
```diff
@@ -70,6 +71,7 @@ func NewSyncerMap(cfg *config.Configuration) map[openrtb_ext.BidderName]usersync
 	insertIntoMap(cfg, syncers, openrtb_ext.BidderSovrn, sovrn.NewSovrnSyncer)
 	insertIntoMap(cfg, syncers, openrtb_ext.BidderSonobi, sonobi.NewSonobiSyncer)
 	insertIntoMap(cfg, syncers, openrtb_ext.BidderYieldmo, yieldmo.NewYieldmoSyncer)
+	insertIntoMap(cfg, syncers, openrtb_ext.BidderVisx, visx.NewVisxSyncer)
 	insertIntoMap(cfg, syncers, openrtb_ext.BidderGamoshi, gamoshi.NewGamoshiSyncer)
 	insertIntoMap(cfg, syncers, openrtb_ext.BidderMgid, mgid.NewMgidSyncer)
```
Comment (guscarreon) [inline usersync/usersyncers/syncer.go]: Add this line here to resolve merge conflicts:
```
      insertIntoMap(cfg, syncers, openrtb_ext.BidderUnruly, unruly.NewUnrulySyncer)
```

**Comment by mk0x9** [comment]
@josephveach @mansinahar @guscarreon 

I have rebased it on top of current master, thanks for notes.
